### ISSUE TITLE
React-Native - No identifers allowed directly after numeric literal

### ISSUE THREAD

**Issue by Tbaut** [polkadot-js/common#1440]
I've upgraded api, util, util-crypto and wasm-crypto, and made sure I import the relevant package at the top level [as per the release note](https://github.com/polkadot-js/wasm/releases/tag/v5.0.1). And I get this runtime error that we've had in the past with BigInt
linked: https://github.com/polkadot-js/common/issues/1253 and https://github.com/polkadot-js/common/issues/1245

I've been looking in the generated bundle for`[0-9]_` or `[0-9]n` but didn't find anything worrying. I'm looking more generally now, but the search has several thousand match.

One syntax I came across that could be problematic is `2e3` here for instance 
```js
          function b() {
            setTimeout(function () {
              return e(t);
            }, 2e3);
 ```
I'll keep searching.
And as usual with RN, no stack trace :roll_eyes: 
my versions:
```
    "@polkadot/api": "^7.13.1",
    "@polkadot/reactnative-identicon": "^1.3.1",
    "@polkadot/util": "8.6.1",
    "@polkadot/util-crypto": "8.6.1",
    "@polkadot/wasm-crypto": "^5.0.1",
```

**Comment by jacogr** [comment]
It is most probably that one - which file/lib?

(there is actually now a post-build step that checks for _ in numbers, so it should be caught. 2e3 however is not checked for)

**Comment by Tbaut** [comment]
Note that there are many `xex`. I've spoted a `xEx` as well

```js
      case 4:
        c = e + 1E4;
        break;

      default:
        c = e + 5E3;
    }
```

The `2e3` is in `node_modules/@polkadot/wasm-crypto-asmjs/cjs/data.js` but I see this notation in many other libs in my node module like prettier for instance..

**Comment by jacogr** [comment]
Where is this from?

```
function b() {
            setTimeout(function () {
              return e(t);
            }, 2e3);
```

Can spot the asm issue, e.g. `$12_1 = $4_1 + 2e3 | 0;` just want to confirm if common is clean before moving the issue...

**Comment by jacogr** [comment]
Can you do a small test with just `const a = 2e3; console.log(a)` before we assume it is certainly that issue?

**Comment by Tbaut** [comment]
I can't find this 2e3 in the setTimeout anywhere else than the bundle.. :shrug: 

Will do the check now.

**Comment by Tbaut** [comment]
2e3 prints "2000" and 3E3 prints "3000" :cry:

**Comment by jacogr** [comment]
Could you comment out the `import '@polkadot/wasm-crypto-asmjs/initAsm'` line - just to see if we can rule that out as a suspect.

**Comment by Tbaut** [comment]
I've added the following to the test project
```
    "@polkadot/util": "^8.6.1",
    "@polkadot/wasm-crypto": "^5.0.1",
```

and imported `import '@polkadot/wasm-crypto/initWasmAsm'` as I do in Stylo, and the app runs without issue. I'll remove the import from Stylo just to double check as well.

**Comment by jacogr** [comment]
Stay with the small one, probably a good place to start...

Test 1 -

```js
import '@polkadot/wasm-crypto/initWasmAsm';

import { waitReady, bip39Generate } from '@polkadot/wasm-crypto';

waitReady().then(() => console.log(bip39Generate(12)));
```

Test 2 -

```js
import '@polkadot/wasm-crypto/initWasmAsm';

import { isHex } from '@polkadot/util';
import { waitReady, bip39Generate } from '@polkadot/wasm-crypto';

waitReady().then(() => console.log(bip39Generate(12)));

console.log(isHex('0x1234');
```

Test 3 -

```js
import '@polkadot/wasm-crypto/initWasmAsm';

import { cryptoWaitReady, blake2AsHex } from '@polkadot/util-crypto';

cryptoWaitReady().then(() => console.log(blake2AsHex('0x1234')));
```

**Comment by Tbaut** [comment]
Test1
```
FATAL: Unable to initialize @polkadot/wasm-crypto
at node_modules/@polkadot/wasm-crypto/init.cjs:35:11 in createPromise
at node_modules/@polkadot/wasm-crypto/init.cjs:43:38 in setPromise
at node_modules/@polkadot/wasm-crypto/bundle.cjs:201:21 in waitReady
at screens/TabOneScreen.tsx:11:39 in TabOneScreen
at node_modules/@react-navigation/native/src/useThenable.tsx:25:13 in resolve

WebAssembly is not available in your environment
at node_modules/@polkadot/util/assert.cjs:37:10 in assert
at node_modules/@polkadot/wasm-crypto/init.cjs:23:0 in createPromise
at node_modules/@polkadot/wasm-crypto/init.cjs:43:38 in setPromise
at node_modules/@polkadot/wasm-crypto/bundle.cjs:201:21 in waitReady
at screens/TabOneScreen.tsx:11:39 in TabOneScreen
at node_modules/@react-navigation/native/src/useThenable.tsx:25:13 in resolve

[Unhandled promise rejection: Error: The WASM interface has not been initialized. Ensure that you wait for the initialization Promise with waitReady() from @polkadot/wasm-crypto (or cryptoWaitReady() from @polkadot/util-crypto) before attempting to use WASM-only interfaces.]
at node_modules/@polkadot/util/assert.cjs:37:10 in assert
at node_modules/@polkadot/wasm-crypto/bridge.cjs:34:21 in <anonymous>
at screens/TabOneScreen.tsx:14:19 in waitReady.then$argument_0

```

Test 2 
```
true

[Unhandled promise rejection: Error: The WASM interface has not been initialized. Ensure that you wait for the initialization Promise with waitReady() from @polkadot/wasm-crypto (or cryptoWaitReady() from @polkadot/util-crypto) before attempting to use WASM-only interfaces.]
at node_modules/@polkadot/util/assert.cjs:37:10 in assert
at node_modules/@polkadot/wasm-crypto/bridge.cjs:31:19 in <global>
```

Test 3
```
FATAL: Unable to initialize @polkadot/wasm-crypto
at node_modules/@polkadot/wasm-crypto/init.cjs:35:11 in createPromise
at node_modules/@polkadot/wasm-crypto/init.cjs:43:38 in setPromise
at node_modules/@polkadot/wasm-crypto/bundle.cjs:201:21 in waitReady
at node_modules/@polkadot/util-crypto/crypto.cjs:18:27 in cryptoWaitReady
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:209215:30 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208854:13 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208586:27 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208530:31 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:193871:56 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at App.tsx:8:0 in <global>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at node_modules/expo/AppEntry.js:3:0 in <global>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at node_modules/metro-runtime/src/polyfills/require.js:201:44 in guardedLoadModule
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:225563:3 in global code

WebAssembly is not available in your environment
at node_modules/@polkadot/util/assert.cjs:37:10 in assert
at node_modules/@polkadot/wasm-crypto/init.cjs:23:0 in createPromise
at node_modules/@polkadot/wasm-crypto/init.cjs:43:38 in setPromise
at node_modules/@polkadot/wasm-crypto/bundle.cjs:201:21 in waitReady
at node_modules/@polkadot/util-crypto/crypto.cjs:18:27 in cryptoWaitReady
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:209215:30 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208854:13 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208586:27 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208530:31 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:193871:56 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at App.tsx:8:0 in <global>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at node_modules/expo/AppEntry.js:3:0 in <global>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at node_modules/metro-runtime/src/polyfills/require.js:201:44 in guardedLoadModule
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:225563:3 in global code

[Unhandled promise rejection: Error: The WASM interface has not been initialized. Ensure that you wait for the initialization Promise with waitReady() from @polkadot/wasm-crypto (or cryptoWaitReady() from @polkadot/util-crypto) before attempting to use WASM-only interfaces.]
at node_modules/@polkadot/util/assert.cjs:37:10 in assert
at node_modules/@polkadot/wasm-crypto/bridge.cjs:34:21 in <anonymous>
at node_modules/@polkadot/util-crypto/helpers.cjs:18:21 in <anonymous>
at screens/TabOneScreen.tsx:13:25 in cryptoWaitReady.then$argument_0

```

**Comment by jacogr** [comment]
Weird that the asm part doesn’t kick in. However, it compiles in all cases.

**Comment by Tbaut** [comment]
I re-ran the second test, and indeed it kicks in, I edited

**Comment by jacogr** [comment]
Ok, then in that test project -

```js
import { ApiPromise, WsProvider } from '@polkadot/api';

ApiPromise
  .create({ provider: new WsProvider('wss://rpc.polkadot.io') })
  .then((api) =>
    api.rpc.chain.subscribeNewHeads((h) => console.log(h.number.unwrap().toHuman()))
  );
```

**Comment by Tbaut** [comment]
So with only the following `@polkadot/*` in my package.json

```
    "@polkadot/api": "^7.13.1",
    "@polkadot/wasm-crypto": "^5.0.1",
```

and the `import '@polkadot/wasm-crypto/initWasmAsm'` in the top level `App.tsx`, and the api test a little lower down the tree I get

```
Android Running app on Android SDK built for x86

FATAL: Unable to initialize @polkadot/wasm-crypto
at node_modules/@polkadot/wasm-crypto/init.cjs:35:11 in createPromise
at node_modules/@polkadot/wasm-crypto/init.cjs:43:38 in setPromise
at node_modules/@polkadot/wasm-crypto/bundle.cjs:201:21 in waitReady
at node_modules/@polkadot/util-crypto/crypto.cjs:18:27 in cryptoWaitReady
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:209612:30 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:209251:13 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:209177:27 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208897:31 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208853:28 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208714:27 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208577:28 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208394:27 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208332:24 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:193673:56 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at App.tsx:9:0 in <global>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at node_modules/expo/AppEntry.js:3:0 in <global>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at node_modules/metro-runtime/src/polyfills/require.js:201:44 in guardedLoadModule
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:278210:3 in global code

WebAssembly is not available in your environment
at node_modules/@polkadot/util/assert.cjs:37:10 in assert
at node_modules/@polkadot/wasm-crypto/init.cjs:23:0 in createPromise
at node_modules/@polkadot/wasm-crypto/init.cjs:43:38 in setPromise
at node_modules/@polkadot/wasm-crypto/bundle.cjs:201:21 in waitReady
at node_modules/@polkadot/util-crypto/crypto.cjs:18:27 in cryptoWaitReady
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:209612:30 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:209251:13 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:209177:27 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208897:31 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208853:28 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208714:27 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208577:28 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208394:27 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:208332:24 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:193673:56 in <unknown>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at App.tsx:9:0 in <global>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at node_modules/expo/AppEntry.js:3:0 in <global>
at node_modules/metro-runtime/src/polyfills/require.js:349:11 in loadModuleImplementation
at node_modules/metro-runtime/src/polyfills/require.js:201:44 in guardedLoadModule
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:278210:3 in global code

Error: The WASM interface has not been initialized. Ensure that you wait for the initialization Promise with waitReady() from @polkadot/wasm-crypto (or cryptoWaitReady() from @polkadot/util-crypto) before attempting to use WASM-only interfaces.
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:172:19 in handleException
at node_modules/react-native/Libraries/Core/setUpErrorHandling.js:24:6 in handleError
at node_modules/@react-native/polyfills/error-guard.js:49:36 in ErrorUtils.reportFatalError
at node_modules/metro-runtime/src/polyfills/require.js:204:6 in guardedLoadModule
at http://192.168.2.4:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false&minify=false:278210:3 in global code

Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called.
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:172:19 in handleException
at node_modules/react-native/Libraries/Core/setUpErrorHandling.js:24:6 in handleError
at node_modules/@react-native/polyfills/error-guard.js:49:36 in ErrorUtils.reportFatalError

```

**Comment by jacogr** [comment]
So it doesn’t seem to stem from there either. (The Wasm init error is still weird though)

**Comment by jacogr** [comment]
So I guess the only thing that is left is the identicon? (No, `color` was not upgraded there...)

**Comment by Tbaut** [comment]
It looks like the wasm error actually gets in the way. I've added the react-identicon (and removed the import api)
```
      <Identicon
        value="5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"
        size={48}
      />
```

and get the same error as above. A weird behavior is that whether or not I have `import '@polkadot/wasm-crypto/initWasmAsm'` the error is always the same "Unable to initialize @polkadot/wasm-crypto".

Not sure what I'm missing here. RN is annoying with caching etc, and I do my best to clear everything every time..

TL;DR
- having `import '@polkadot/wasm-crypto/initWasmAsm'`  doesn't make a diff, no error with an empty project and the import
- using anything like the api or react-identicon throws the wasm-crypto not initialized

**Comment by jacogr** [comment]
There could be an issue with `initWasmAsm`, try the `initOnlyAsm` part. (But we should move that to the wasm repo to dive through... something is weird...)

But anyway, all of this and no literal error?

**Comment by Tbaut** [comment]
I tried with `initOnlyAsm` and everything as https://github.com/polkadot-js/common/issues/1440#issuecomment-1076224768 holds

**Comment by jacogr** [comment]
Ok, so no numeric literal issues?

**Comment by Tbaut** [comment]
> Ok, so no numeric literal issues?

nope

**Comment by Tbaut** [comment]
sorry, it took me a while to get back to this.
Test from https://github.com/polkadot-js/common/issues/1440#issuecomment-1075457274 1, 2 and 3 succeed now :tada: .

so I went on with testing the Identicon, and I got a new error that I've never seen

```bash
TypeError: undefined is not an object (evaluating '_$$_REQUIRE(_dependencyMap[4], "@polkadot/react-identicon").Identicon')

This error is located at:
    in App (created by ExpoRoot)
    in ExpoRoot
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
TypeError: undefined is not an object (evaluating '_styledComponents.default.div.withConfig')
Error: Requiring module "node_modules/@polkadot/react-identicon/index.cjs", which threw an exception: TypeError: undefined is not an object (evaluating '_styledComponents.default.div.withConfig')
```

**Comment by jacogr** [comment]
The import should be -

```js
import Identicon from '@polkadot/reactnative-identicon';
```

**Comment by Tbaut** [comment]
arg, that was a dumb one sorry about that. Importing the default export didn't show the identicon, or anything other that should appear on the screen like a "hello" text, the console.log didn't print either. But there was no error either. I'll dig in the next days with both yours, and my test project.

**Comment by Tbaut** [comment]
With my test project I ended up installing and using all of these without ever seeing the numeric literal issue.

-  :heavy_check_mark: `@polkadot/react-native-identicon` (needs `react-native-svg` and eventually `react-native-svg-transform`)
-  :heavy_check_mark: `@polkadot/api`
- :heavy_check_mark: `@polkadot/util`
- :heavy_check_mark: `@polkadot/util-crypto`
- :heavy_check_mark: `@polkadot/wasm-crypto`

I went back to Stylo, upgraded the polkadot deps and added the resolution for each of them. I don't have the issue any more either (but I'm fighting against the "WASM interface has not been initialized" one :sob: )

**Comment by jacogr** [comment]
On the first one, I had issues in my test project.

**Comment by jacogr** [comment]
So the change in the latest crypto deps was that we did this automatically -

```js
cryptoWaitReady().catch(console.log);
```

Now it requires it explicitly. I would almost suggest the following -

```js
import `@polkadot/wasm-crypto/initOnlyAsm';
import { cryptoWaitReady } from '@polkadot/util-crypto';
import { registerRootComponent } from 'expo';

import App from './App';

 cryptoWaitReady().catch(console.log);

// registerRootComponent calls AppRegistry.registerComponent('main', () => App);
// It also ensures that whether you load the app in Expo Go or in a native build,
// the environment is set up appropriately
registerRootComponent(App);
```
 
 (And then still have the async version to kickstart in your peoject, as per normal). The above just replicates the old kickstart.

**Comment by Tbaut** [comment]
Happy to open a new issue for this. I can't get out of this issue, I pushed the code here
https://github.com/stylo-app/stylo/compare/wasm-not-ready?expand=1

Doing as you advise, in the `index.js` and having an `await cryptoWaitReady().catch(console.log)` before any use of `util-crypto` function. I get no error logged from the catch it seems

```
 BUNDLE  ./index.android.js 

 WARN  Require cycle: node_modules/react-native-crypto/index.js -> node_modules/react-native-randombytes/index.js -> node_modules/sjcl/sjcl.js -> node_modules/react-native-crypto/index.js

Require cycles are allowed, but can result in uninitialized values. Consider refactoring to remove the need for a cycle.
 ERROR  Error: The WASM interface has not been initialized. Ensure that you wait for the initialization Promise with waitReady() from @polkadot/wasm-crypto (or cryptoWaitReady() from @polkadot/util-crypto) before attempting to use WASM-only interfaces.
 ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.
      This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.
 ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.
      This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.

```
